server/s3_post_complete_action.cc

```diff
@@ -1048,6 +1049,22 @@ void S3PostCompleteAction::delete_old_object_success() {
 
 void S3PostCompleteAction::remove_old_object_version_metadata() {
   s3_log(S3_LOG_INFO, stripped_request_id, "%s Entry\n", __func__);
+  if (object_metadata) {
+    // printf("object name from multipart : ");
+    // printf(multipart_metadata->get_object_name());
+    // printf("object name from request : ");
+    // printf(request->get_object_name());
+    assert(multipart_metadata->get_object_name() == request->get_object_name());
+    object_metadata->set_oid(old_object_oid);
+    object_metadata->set_layout_id(old_layout_id);
+    object_metadata->set_version_id(
+        multipart_metadata->get_old_obj_version_id());
+    object_metadata->remove_version_metadata(
+        std::bind(&S3PostCompleteAction::remove_old_fragments, this),
+        std::bind(&S3PostCompleteAction::remove_old_fragments, this));
+  }
+  s3_log(S3_LOG_DEBUG, "", "%s Exit", __func__);
+}
```

Comment:
Is this the end of function S3PostCompleteAction::remove_old_object_version_metadata()? Then check if you really need code from line #1069 to line#1081 (see below code)